src/Microsoft.DotNet.ProjectModel/DependencyContextBuilder.cs

```diff
@@ -48,16 +50,38 @@ public DependencyContextBuilder(string referenceAssembliesPath)
             var compilationOptions = compilerOptions != null
                 ? GetCompilationOptions(compilerOptions)
                 : CompilationOptions.Default;
+
+            var runtimeSignature = GenerateRuntimeSignature(runtimeExports);
+
             return new DependencyContext(
-                target.DotNetFrameworkName,
-                runtime,
-                portable,
+                new TargetInfo(target.DotNetFrameworkName, runtime, runtimeSignature, portable),
                 compilationOptions,
                 GetLibraries(compilationExports, dependencyLookup, runtime: false).Cast<CompilationLibrary>(),
                 GetLibraries(runtimeExports, dependencyLookup, runtime: true).Cast<RuntimeLibrary>(),
                 new RuntimeFallbacks[] {});
         }
 
+        private static string GenerateRuntimeSignature(IEnumerable<LibraryExport> runtimeExports)
+        {
+            var sha1 = SHA1.Create();
+            var builder = new StringBuilder();
+            var packages = runtimeExports
+                .Where(libraryExport => libraryExport.Library.Identity.Type == LibraryType.Package);
+            foreach (var libraryExport in packages)
+            {
+                builder.Append(libraryExport.Library.Identity.Name);
+                builder.Append(libraryExport.Library.Identity.Version);
```

Comment:
Could you add a separator between each individual library, just to be safe? A newline would be sufficient, so just changing this to `AppendLine`